opentelemetry-sdk/src/opentelemetry/sdk/metrics/__init__.py

```diff
@@ -536,13 +567,12 @@ def register_valueobserver(
             label_keys,
             enabled,
         )
-        with self.observers_lock:
-            self.observers.add(ob)
+        self._register_observer(ob)
         return ob
 
     def unregister_observer(self, observer: metrics_api.Observer) -> None:
         with self.observers_lock:
-            self.observers.remove(observer)
+            self.observers.pop(observer.name, None)
```

Comment:
Looks like an exception was raised here previously. Any reason to add in the `None` default?

Reply:
Not sure why I did that, changed.